### PR TITLE
Add new paid stats site features: STATS_FREE, STATS_COMMERCIAL

### PR DESCRIPTION
**Pull request by lsl**
Related to https://github.com/Automattic/red-team/issues/297

## Proposed Changes

* Adds new paid stats site features: STATS_FREE, STATS_COMMERCIAL
* Updates shouldGateStats to make use of the new site features

TODO:
- [x] We still need to add something to handle 2024-09-01 - 2024-12-tbd free sites, probably STATS_BASIC p1733289129342149/1733176661.084479-slack-C068X5XJT3J

## Testing Instructions

Test with / without the flag enabled:

```
?flags=stats/paid-wpcom-v3 (with, default)
?flags=-stats/paid-wpcom-v3 (without)
```

Test with / without the diff applied to your sandbox:

D167647-code

Test sites:
  - Free before 2024-01-09 (Should see v2 gating on UTM/Devices only - shouldn't actually see these at all (yet))
  - Free after 2024-01-09 (Should see v2 gating)
  - Free after 2024-12-tbd (Should see v3 gating)
  - Personal before 2024-01-09 (Should see v2 gating on UTM/Devices only)
  - Personal after 2024-01-09 (Should see less v2 gating - clicks+referrers should not be gated)
  - Premium before and after 2024-01-09 (Should see no gating)

Remember to test Jetpack/Atomic sites too, will need https://github.com/Automattic/jetpack/pull/40421

Unit tests:
```
yarn run test-client client/my-sites/stats/hooks/test/use-should-gate-stats.ts
```
(Should probably improve these now, I've only made them work with the new stats break down)

E2E:
```
yarn workspace wp-e2e-tests decrypt-secrets (only if you need to)
yarn workspace wp-e2e-tests test -- test/e2e/specs/stats/stats.ts
```

### CODE REVIEW

File: client/my-sites/stats/hooks/use-should-gate-stats.ts [hunk 1]
```diff
@@ -78,17 +82,37 @@ const granularControlForJetpackStatsCommercialPaywall = [
...
+	STATS_FEATURE_SUMMARY_LINKS_QUARTER,
+	STATS_FEATURE_SUMMARY_LINKS_YEAR,
+	STATS_FEATURE_SUMMARY_LINKS_ALL,
+	STATS_FEATURE_UTM_STATS,
 ];
 
-// Gated controls for WPCOM sites without the FEATURE_STATS_PAID feature.
-const granularControlForPaidStats = [
+// wpcom: Gate UTM and device stats for sites with STATS_FREE feature, this is the feature applied to legacy sites.
+const freeStats = [ STATS_FEATURE_UTM_STATS, STATS_TYPE_DEVICE_STATS ];
```
Comment (dognose24) [inline client/my-sites/stats/hooks/use-should-gate-stats.ts]: As the feature `stats-free` seems unnecessary, should we change its name to `stats-legacy-free` to highlight the check here?

Comment (lsl) [reply client/my-sites/stats/hooks/use-should-gate-stats.ts]: STATS_FREE is a Jetpack site feature I'm re-using here so I don't think the rename makes sense. But maybe using the Jetpack site feature here doesn't make sense either.

**Comment by Initsogar** [comment]
I will post my test results as soon as I have them.
I started with simple sites, and all scenarios seem to be good!

## Definitions (my understanding)
* **v2 gating**: limited to 7 days. All modules gated, except Post & pages, Locations, and File downloads.
* **v3 gating**: limited to 7 days. All modules gated, including Insights page.

## Simple sites ✅

| Site | Plan | Scenario | Expected gating behavior | Test result | Notes |
|---|---|---|---|---|---|
| initsogar.* | Free | Created before 2024-01-09 | v2 gating on UTM/Devices only | ✅ | No gates, full acess. UTM/Devices not available. |
| initsogarfree.* | Free | Created after 2024-01-09 | v2 gating | ✅ | |
| initsogarstatsbasic.* | Free | Created after 2024-12-TBD | v3 gating | ✅ | I needed to update the `after` date in the features API to reproduce it. |
| initsogar7.* | Personal | Created before 2024-01-09 | v2 gating on UTM/Devices only | ✅ | No gates, full acess. UTM/Devices not available. |
| initsogarpersonal.* | Personal | Created after 2024-01-09 | Less v2 gating (clicks + referrers not gated) | ✅ | It only has Authors and Search terms gated. Clicks and referrers are not gated. |
| initsogarpremium.* | Premium | Any date | No gating | ✅ | stats-paid and stats-commercial |

**Comment by Initsogar** [comment]
## Atomic sites ✅
All sites tested using Calypso Blue and wp-admin.

**Setup:**
* Run `install-plugin.sh odyssey-stats update/stats-newfeatures` on sandbox
* Point `widgets.wp.com` to my sandbox
* For wp-admin: ensured I'm using the classic style admin interface in `General > Settings`

| Site | Plan | Scenario | Expected gating behavior | Test result | Notes |
|---|---|---|---|---|---|
| floowandereeze.* | Business | with the WPCOM diff | No gating | ✅ | |
| initsogar36.* | Business | without the WPCOM diff | No gating | ✅ | |

**Comment by kangzj** [comment]
I tested Business and up (Atomic sites) in Calypso + Odyssey Stats. Everything works as expected, with no modules gated. However, the UTM and Device stats are still hidden, which is another issue to solve.

**Comment by Initsogar** [comment]
## Jetpack sites ✅

I tested with the same self-hosted site these scenarios. Let me know if I should test another one.

**Setup:**
* Run `install-plugin.sh odyssey-stats update/stats-newfeatures` on sandbox
* Point `widgets.wp.com` to my sandbox
* Point my self-hosted site to my WPCOM sandbox
* Apply the WPCOM diff when needed
* Add/remove subscriptions to refresh features cache and properly test the different scenarios
 
| Plan | Scenario | Expected gating behavior | Test result | Notes |
|---|---|---|---|---|
| Free | with the WPCOM diff | v2 gating on UTM/Devices only | ✅ | stats-paid. UTM/Devices gated. |
| Jetpack Complete | with the WPCOM diff | No gating | ✅ | stats-paid + stats-commercial |
| Free | without the WPCOM diff | v2 gating on UTM/Devices only | ✅ | No feature. UTM/Devices gated. |
| Jetpack Complete | without the WPCOM diff | No gating | ✅ | stats-paid |

**Comment by kangzj** [comment]
## Jetpack sites - Commercial ✅

I tested a commercial site both in Calypso and Odyssey.
 
| Plan | Scenario | Expected gating behavior | Test result | Notes |
|---|---|---|---|---|
| Free / passed grace period | with the WPCOM diff | restricted dashboard | ✅ | only 7-day traffic chart available |
| Free / passed grace period | without the WPCOM diff | restricted dashboard | ✅ | only 7-day traffic chart available |
| Free / before grace period | with the WPCOM diff | Personal stats | ✅ | All stats except for UTM and device |
| Free / before grace period | with the WPCOM diff | Personal stats | ✅ | All stats except for UTM and device |


Combined with the test result for personal sites @Initsogar finished [above](https://github.com/Automattic/wp-calypso/pull/97041#issuecomment-2521924203). We can state that the diff + PR don't affect the existing upsells for Jetpack self-hosted sites.

**Comment by Initsogar** [comment]
## Simple sites - without the WPCOM diff ❌

Setup:
* Testing without the `stats/paid-wpcom-v3` flag to ensure users are not negatively impacted by this change. (passing `?flags=-stats/paid-wpcom-v3`).
* Comparing the results from Calypso Live Branch vs WordPress.com production (non-proxied).

| Site | Plan | Scenario | Expected gating behavior | Test result | Notes |
|---|---|---|---|---|---|
| initsogar.* | Free | Created before 2024-01-09 | v2 gating on UTM/Devices only | ❌ | stats-paid. Authors & Search terms gated. UTM/Devices not available. |
| initsogarfree.* | Free | Created after 2024-01-09 | v2 gating | ✅ | No feature. Limited to 7 days. All modules gated, except Post & pages, Locations, and File downloads. |
| initsogarstatsbasic.* | Free | Created after 2024-12-TBD | v2 gating | ✅ | No feature. Similar to previous scenario. Since the `after` date is implemented on the WPCOM diff. |
| initsogar7.* | Personal | Created before 2024-01-09 | v2 gating on UTM/Devices only | ✅ | stats-paid. Authors & Search terms gated. UTM/Devices not available. |
| initsogarpersonal.* | Personal | Created after 2024-01-09 | v2 gating | ✅ | It only has Authors and Search terms gated. Clicks and referrers are not gated. |
| initsogarpremium.* | Premium | Any date | No gating | ❌ | stats-paid. Authors & Search terms gated. UTM/Devices not available. |

Additional detail about failing scenarios:

| Site | Plan | Scenario | Features without diff | Features with diff |
|---|---|---|---|---|
| initsogar.* | Free | Created before 2024-01-09 | stats-paid | stats-free |
| initsogarpremium.* | Premium | Any date | stats-paid | stats-basic, stats-paid, stats-commercial |

**Review by Initsogar — APPROVED** [review]
I ran a new set of tests, and everything seems great. I’m still not sure how to test the Free/passed grace period scenario for Jetpack commercial sites, but based on the current results, it should be good as well. cc/ @kangzj 

## Simple sites - without the WPCOM diff and without v3 flag
> [!NOTE]
> Free and Personal sites before 2024-01-09 are not gated without the WPCOM diff and without the `stats/paid-wpcom-v3` flag, but it will be gated once the flag is enabled (and the diff is deployed).

| Site | Plan | Scenario | Expected gating behavior | Test result |
|---|---|---|---|---|
| initsogar.* | Free | Created before 2024-01-09 | No gating (temporarily for legacy sites) | ✅ |
| initsogarfree.* | Free | Created after 2024-01-09 | v2 gating | ✅ |
| initsogarstatsbasic.* | Free | Created after 2024-12-TBD | v2 gating | ✅ |
| initsogar7.* | Personal | Created before 2024-01-09 | No gating (temporarily for legacy sites)| ✅ |
| initsogarpersonal.* | Personal | Created after 2024-01-09 | v2 gating | ✅ |
| initsogarpremium.* | Premium | Any date | No gating | ✅ |

## Simple sites - with the WPCOM diff and v3 flag enabled

| Site | Plan | Scenario | Expected gating behavior | Test result | Features |
|---|---|---|---|---|---|
| initsogar.* | Free | Created before 2024-01-09 | UTM & Devices gated | ✅ | stats-free|stats-basic |
| initsogarfree.* | Free | Created after 2024-01-09 | v2 gating | ✅ | stats-basic |
| initsogarstatsbasic.* | Free | Created after 2024-12-TBD | v3 gating | ✅ | no stats feature |
| initsogar7.* | Personal | Created before 2024-01-09 | UTM & Devices gated | ✅ | stats-free|stats-basic|stats-paid |
| initsogarpersonal.* | Personal | Created after 2024-01-09 | v2 gating | ✅ | stats-paid |
| initsogarpremium.* | Premium | Any date | No gating | ✅ | stats-paid|stats-commercial |

## Atomic sites

| Site | Plan | Scenario | Expected gating behavior | Test result |
|---|---|---|---|---|
| floowandereeze.* | Business | with the WPCOM diff | No gating | ✅ |
| initsogar36.* | Business | without the WPCOM diff | No gating | ✅ |

## Jetpack sites - Non-commercial ✅

I tested with the same self-hosted site these scenarios. Let me know if I should test another one.

**Setup:**
* Run `install-plugin.sh odyssey-stats update/stats-newfeatures` on sandbox
* Point `widgets.wp.com` to my sandbox
* Point my self-hosted site to my WPCOM sandbox
* Apply the WPCOM diff when needed
* Add/remove subscriptions to refresh features cache and properly test the different scenarios
 
| Plan | Scenario | Expected gating behavior | Test result | Notes |
|---|---|---|---|---|
| Free | without the WPCOM diff | UTM & Devices gated | ✅ | No feature. UTM/Devices gated. |
| Free | with the WPCOM diff | UTM & Devices gated | ✅ | stats-paid. UTM/Devices gated. |
| Jetpack Complete | without the WPCOM diff | No gating |  | stats-paid |
| Jetpack Complete | with the WPCOM diff | No gating | ✅ | stats-paid|stats-commercial |

## Jetpack sites - Commercial ✅

| Plan | Scenario | Expected gating behavior | Test result | Notes |
|---|---|---|---|---|
| Free / passed grace period | with the WPCOM diff | restricted dashboard | ⌛️ | |
| Free / passed grace period | without the WPCOM diff | restricted dashboard | ✅ | only 7-day traffic chart available |
| Free / before grace period | with the WPCOM diff | Personal stats | ✅ | All stats except for UTM and device |
| Free / before grace period | without the WPCOM diff | Personal stats | ✅ | All stats except for UTM and device |

**Comment by kangzj** [comment]
I tested this scenario, and it's a pass ✅ 

Free / passed grace period |	with the WPCOM diff |	restricted dashboard  |	only 7-day traffic chart available ✅

**Pull request merged**